### PR TITLE
Enable ruff's EM (flake8-errmsg) rules and fix violations

### PR DESCRIPTION
**Pull request by seisman**
**Description of proposed changes**

This PR enables ruff's [`EM`](https://docs.astral.sh/ruff/rules/#flake8-errmsg-em) rules and fixes some violations.

Address #2741.

### CODE REVIEW

File: pygmt/io.py [hunk 1]
```diff
@@ -38,7 +38,8 @@ def load_dataarray(filename_or_obj, **kwargs):
     xarray.open_dataarray
     """
     if "cache" in kwargs:
-        raise TypeError("cache has no effect in this context")
+        msg = "cache has no effect in this context"
```
Comment (michaelgrund) [inline pygmt/io.py:41]:
```suggestion
        msg = "Cache has no effect in this context."
```

File: pygmt/src/meca.py [hunk 2]
```diff
@@ -424,7 +424,8 @@ def meca(  # noqa: PLR0912, PLR0913, PLR0915
             )
     elif isinstance(spec, np.ndarray):  # spec is a numpy array
         if convention is None:
-            raise GMTInvalidInput("'convention' must be specified for an array input.")
+            msg = "'convention' must be specified for an array input."
```
Comment (michaelgrund) [inline pygmt/src/meca.py:427]:
```suggestion
            msg = "'Convention' must be specified for an array input."
```

File: pygmt/clib/session.py [hunk 3]
```diff
@@ -1856,9 +1861,8 @@ def virtualfile_in(
             elif check_kind == "vector":
                 valid_kinds += ("empty", "matrix", "vectors", "geojson")
             if kind not in valid_kinds:
-                raise GMTInvalidInput(
-                    f"Unrecognized data type for {check_kind}: {type(data)}"
-                )
+                msg = f"Unrecognized data type for {check_kind}: {type(data)}"
```
Comment (michaelgrund) [inline pygmt/clib/session.py:1864]:
```suggestion
                msg = f"Unrecognized data type for {check_kind}: {type(data)}."
```

**Review by michaelgrund — APPROVED** [review]

File: pygmt/src/meca.py (hunk 2 above)
Comment (seisman) [reply pygmt/src/meca.py:427]: It should be lower case since it's a parameter name in quotes.

File: pygmt/datasets/earth_relief.py [hunk 4]
```diff
@@ -155,17 +155,19 @@ def load_earth_relief(
...
         )
+        raise GMTInvalidInput(msg)
     # Use SRTM or not.
     if use_srtm and resolution in land_only_srtm_resolutions:
         if data_source != "igpp":
-            raise GMTInvalidInput(
+            msg = (
                 f"Option 'use_srtm=True' doesn't work with data source '{data_source}'."
                 " Please set 'data_source' to 'igpp'."
             )
```
Comment (yvonnefroehlich) [inline pygmt/datasets/earth_relief.py:169]:
```suggestion
            msg = (
                f"Option 'use_srtm=True' doesn't work with data source '{data_source}'. "
                "Please set 'data_source' to 'igpp'."
            )
```

File: pygmt/src/grd2xyz.py [hunk 5]
```diff
@@ -144,11 +144,11 @@ def grd2xyz(
     output_type = validate_output_table_type(output_type, outfile=outfile)
 
     if kwargs.get("o") is not None and output_type == "pandas":
-        raise GMTInvalidInput(
+        msg = (
             "If 'outcols' is specified, 'output_type' must be either 'numpy'"
             "or 'file'."
         )
```
Comment (yvonnefroehlich) [inline pygmt/src/grd2xyz.py:150]:
```suggestion
        msg = (
            "If 'outcols' is specified, 'output_type' must be either 'numpy' "
            "or 'file'."
        )
```

File: pygmt/src/grd2xyz.py [hunk 6]
```diff
@@ -144,11 +144,11 @@ def grd2xyz(
...
-
+        raise GMTInvalidInput(msg)
     # Set the default column names for the pandas dataframe header.
     column_names: list[str] = ["x", "y", "z"]
     # Let output pandas column names match input DataArray dimension names
```
Comment (yvonnefroehlich) [inline pygmt/src/grd2xyz.py:154]:
```suggestion
    # Set the default column names for the pandas DataFrame header.
    column_names: list[str] = ["x", "y", "z"]
    # Let output pandas column names match input DataArray dimension names
```

File: pygmt/src/plot.py [hunk 7]
```diff
@@ -246,7 +246,8 @@ def plot(
             ("symbol", symbol),
         ]:
             if is_nonstr_iter(value):
-                raise GMTInvalidInput(f"'{name}' can't be 1-D array if 'data' is used.")
+                msg = f"'{name}' can't be 1-D array if 'data' is used."
```
Comment (yvonnefroehlich) [inline pygmt/src/plot.py:249]:
```suggestion
                msg = f"'{name}' can't be a 1-D array if 'data' is used."
```

File: pygmt/src/plot3d.py [hunk 8]
```diff
@@ -230,7 +230,8 @@ def plot3d(
             ("symbol", symbol),
         ]:
             if is_nonstr_iter(value):
-                raise GMTInvalidInput(f"'{name}' can't be 1-D array if 'data' is used.")
+                msg = f"'{name}' can't be 1-D array if 'data' is used."
```
Comment (yvonnefroehlich) [inline pygmt/src/plot3d.py:233]:
```suggestion
                msg = f"'{name}' can't be a 1-D array if 'data' is used."
```

File: pygmt/src/project.py [hunk 9]
```diff
@@ -222,15 +222,14 @@ def project(
           (depends on ``output_type``)
     """
     if kwargs.get("C") is None:
-        raise GMTInvalidInput("The `center` parameter must be specified.")
+        msg = "The `center` parameter must be specified."
```
Comment (yvonnefroehlich) [inline pygmt/src/project.py:225]:
```suggestion
        msg = "The 'center' parameter must be specified."
```

File: pygmt/src/project.py [hunk 10]
```diff
@@ -222,15 +222,14 @@ def project(
...
+        raise GMTInvalidInput(msg)
     if kwargs.get("G") is None and data is None:
-        raise GMTInvalidInput(
-            "The `data` parameter must be specified unless `generate` is used."
-        )
+        msg = "The `data` parameter must be specified unless `generate` is used."
```
Comment (yvonnefroehlich) [inline pygmt/src/project.py:228]:
```suggestion
        msg = "The 'data' parameter must be specified unless 'generate' is used."
```

File: pygmt/src/project.py [hunk 11]
```diff
@@ -222,15 +222,14 @@ def project(
...
+        raise GMTInvalidInput(msg)
     if kwargs.get("G") is not None and kwargs.get("F") is not None:
-        raise GMTInvalidInput(
-            "The `convention` parameter is not allowed with `generate`."
-        )
+        msg = "The `convention` parameter is not allowed with `generate`."
```
Comment (yvonnefroehlich) [inline pygmt/src/project.py:231]:
```suggestion
        msg = "The 'convention' parameter is not allowed with 'generate'."
```

File: pygmt/src/text.py [hunk 12]
```diff
@@ -187,17 +187,21 @@ def text_(  # noqa: PLR0912
         + (position is not None)
         + (x is not None or y is not None)
     ) != 1:
-        raise GMTInvalidInput("Provide either textfiles, x/y/text, or position/text.")
+        msg = "Provide either textfiles, x/y/text, or position/text."
```
Comment (yvonnefroehlich) [inline pygmt/src/text.py:190]:
```suggestion
        msg = "Provide either 'textfiles', 'x'/'y'/'text', or 'position'/'text'."
```

File: pygmt/src/velo.py [hunk 13]
```diff
@@ -243,15 +243,15 @@ def velo(self, data=None, **kwargs):
     if kwargs.get("S") is None or (
         kwargs.get("S") is not None and not isinstance(kwargs["S"], str)
     ):
-        raise GMTInvalidInput(
-            "The parameter `spec` is required and has to be a string."
-        )
+        msg = "The parameter `spec` is required and has to be a string."
```
Comment (yvonnefroehlich) [inline pygmt/src/velo.py:246]:
```suggestion
        msg = "The parameter 'spec' is required and has to be a string."
```

File: pygmt/src/grdtrack.py [hunk 14]
```diff
@@ -290,13 +290,16 @@ def grdtrack(
...
+        raise GMTInvalidInput(msg)
 
     if points is None and kwargs.get("E") is None:
-        raise GMTInvalidInput("Must give 'points' or set 'profile'.")
+        msg = "Must give 'points' or set 'profile'."
+        raise GMTInvalidInput(msg)
 
     if hasattr(points, "columns") and newcolname is None:
-        raise GMTInvalidInput("Please pass in a str to 'newcolname'")
+        msg = "Please pass in a str to 'newcolname'"
```
Comment (yvonnefroehlich) [inline pygmt/src/grdtrack.py:301]:
```suggestion
        msg = "Please pass in a str to 'newcolname'."
```

File: pygmt/src/grd2cpt.py [hunk 15]
```diff
@@ -182,7 +182,8 @@ def grd2cpt(grid, **kwargs):
     >>> fig.show()
     """
     if kwargs.get("W") is not None and kwargs.get("Ww") is not None:
-        raise GMTInvalidInput("Set only categorical or cyclic to True, not both.")
+        msg = "Set only categorical or cyclic to True, not both."
```
Comment (yvonnefroehlich) [inline pygmt/src/grd2cpt.py:185]:
```suggestion
        msg = "Set only 'categorical' or 'cyclic' to True, not both."
```

File: pygmt/src/x2sys_cross.py [hunk 16]
```diff
@@ -210,7 +210,8 @@ def x2sys_cross(
                 # Save pandas.DataFrame track data to temporary file
                 file_contexts.append(tempfile_from_dftrack(track=track, suffix=suffix))
             case _:
-                raise GMTInvalidInput(f"Unrecognized data type: {type(track)}")
+                msg = f"Unrecognized data type: {type(track)}"
```
Comment (seisman) [inline pygmt/src/x2sys_cross.py:213]:
```suggestion
                msg = f"Unrecognized data type: {type(track)}."
```

File: pygmt/tests/test_clib_loading.py [hunk 17]
```diff
@@ -133,7 +133,8 @@ def _mock_ctypes_cdll_return(self, libname):
         if isinstance(libname, str):
             # libname is an invalid library path in string type,
             # raise OSError like the original ctypes.CDLL
-            raise OSError(f"Unable to find '{libname}'")
+            msg = f"Unable to find '{libname}'"
```
Comment (seisman) [inline pygmt/tests/test_clib_loading.py:136]:
```suggestion
            msg = f"Unable to find '{libname}'."
```

**Pull request merged**